### PR TITLE
Donate page

### PR DESCRIPTION
**Pull request by egiurleo**
This PR implements a new page that allows visitors to donate to WNB.rb!

Screen Shot 2022-02-11 at 2 48 46 PM

I want to explain a little bit why the page works the way it does: we use Stripe to generate the checkout pages. From the website UI, Stripe does not allow us to create a checkout page where the user can choose their own price. In order to allow the user to choose any donation amount, we would have to use the Stripe API, which would take longer to develop. So, as a workaround, we generated the checkout pages for 8 prices, then included links to all of them on this page. In the future, if someone wants to implement "Other" functionality, that would be a welcome contribution.

### CODE REVIEW

File: app/javascript/components/pages/Donate.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        </SharedLayout>
+    );
+};
+
+const DonationAmounts = ({ selectedPrice, setSelectedPrice }) => {
+    const prices = useMemo(() => {
+        return process.env.NODE_ENV === 'development'
+            ? testDonationAmounts
+            : productionDonationAmounts;
+    }, []);
```
Comment (egiurleo) [inline app/javascript/components/pages/Donate.jsx:52]: @stephanieminn I'd like your feedback on this pattern -- there are two sets of donation links, test links and real links. The test links prevent us from polluting our real payment data when making test payments. I've set this up by creating two JavaScript objects and here I'm using the env to determine which object to use as the prices. Does this... make sense?

File: app/javascript/components/pages/Donate.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                {prices.map((price) => {
+                    return (
+                        <button
+                            key={price.value}
+                            className={`donation-amount ${
+                                selectedPrice === price.value ? 'selected' : null
+                            }`}
+                            onClick={() => setSelectedPrice(price.value)}
+                        >
+                            ${price.value.toLocaleString()}
```
Comment (egiurleo) [inline app/javascript/components/pages/Donate.jsx:72]: @jonsamp taught me about `toLocaleString`, which formats the price number correctly (e.g. 1000 becomes 1,000).

File: app/javascript/datasources/index.js [hunk 3]
```diff
@@ -48,3 +48,25 @@ export const postJobsAuthenticate = async (password) => {
 
     return await result.json();
 };
+
+export const testDonationAmounts = [
```
Comment (egiurleo) [inline app/javascript/datasources/index.js:52]: @jemmaissroff I think we originally said 10, 20, 50, 100... but an even number of prices looks better than an odd one so I switched it to 10, 25, 50, 75, 100... Let me know if that looks okay to you.

File: app/javascript/components/pages/Donate.jsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+                    <p className="mb-5">
+                        Donating to WNB.rb enables us to invest more resources in supporting women
+                        and non-binary people in the Ruby community.
+                    </p>
+                    <p>
+                        Here are some of the initiatives we plan to launch in 2022 with your help:
+                    </p>
+                    <ul className="list-disc ml-10">
+                        <li>Community fund for educational materials</li>
+                        <li>Full scholarships to RubyConf and RailsConf</li>
```
Comment (jemmaissroff) [inline app/javascript/components/pages/Donate.jsx:37]: Maybe "Travel and housing scholarships" - I don't think I would understand how this is different to the existing railsconf scholarships if I didn't already know

**Comment by egiurleo** [comment]
@jemmaissroff Thanks for the comments! I addressed them and pushed up my changes.

**Review by jemmaissroff — APPROVED** [review]

**Pull request merged**